src/relations/db.py

```diff
@@ -239,9 +241,23 @@ def _on_relation_joined(self, join_event: RelationJoinedEvent):
         dbs = self.charm.generate_relation_databases()
         dbs[str(join_event.relation.id)] = {"name": database, "legacy": True}
         if self.admin:
-            dbs["*"] = {"name": "*", "auth_dbname": database}
+            dbs["*"] = {"name": "*", "auth_dbname": database, "legacy": False}
```

Comment:
Legacy flag is added for backwards compatibility, adding it from the get go so that the hash remains consistent.